src/AppContainer/index.tsx

```diff
@@ -1,49 +1,93 @@
-import React, { useEffect, useState, useRef, useContext } from 'react';
+import React, { useEffect, useState, useRef, useContext, useCallback } from 'react';
 import { AppState } from 'react-native';
 import { AxiosRequestConfig } from 'axios';
 import axiosNotLogged from '../api/axios/notLogged';
 import axiosLogged from '../api/axios/logged';
 import Version from '../api/Versions';
+import { Context as AuthContext } from '../context/AuthContext';
 import { ACTIVE_STATE } from '../core/data/constants';
+import asyncStorage from '../core/helpers/asyncStorage';
 import UpdateAppModal from '../components/modals/UpdateAppModal';
 import MaintenanceModal from '../components/modals/MaintenanceModal';
 import AppNavigation from '../navigation/AppNavigation';
-import { Context as AuthContext } from '../context/AuthContext';
 
 const AppContainer = () => {
   const [updateAppVisible, setUpdateAppVisible] = useState<boolean>(false);
   const [maintenanceModaleVisible, setMaintenanceModalVisible] = useState<boolean>(false);
   const [axiosInitialized, setAxiosInitialized] = useState<boolean>(false);
-  const loggedAxiosInterceptorId = useRef<number | null>(null);
-  const { refreshLoggedUser, companiToken } = useContext(AuthContext);
+  const axiosLoggedRequestInterceptorId = useRef<number | null>(null);
+  const axiosLoggedResponseInterceptorId = useRef<number | null>(null);
+  const { refreshLoggedUser, companiToken, signOut, refreshCompaniToken } = useContext(AuthContext);
+
+  const handleUnauthorizedRequest = useCallback(async (error) => {
+    const {
+      companiToken: oldCompaniToken,
+      companiTokenExpireDate: oldCompaniTokenExpireDate,
+    } = await asyncStorage.getCompaniToken();
+    if (asyncStorage.isTokenValid(oldCompaniToken, oldCompaniTokenExpireDate)) {
+      await signOut();
+      return Promise.reject(error);
+    } // handle invalid refreshToken reception from api wich trigger infite 401 calls
+
+    await asyncStorage.removeCompaniToken();
+    const { refreshToken } = await asyncStorage.getRefreshToken();
+    await refreshCompaniToken(refreshToken);
+
+    const { companiToken: newCompaniToken, companiTokenExpireDate } = await asyncStorage.getCompaniToken();
+
+    if (asyncStorage.isTokenValid(newCompaniToken, companiTokenExpireDate)) {
+      const config = { ...error.config };
+      config.headers['x-access-token'] = newCompaniToken;
```

Comment:
tu as pu trouver finalement la difference entre headers et headers.common ?

Reply:
On a eu l'impression avec Kenny que ce qu'on mettait dans common definissait le header pour toutes les prochaines requetes, alors que dans headers seulement pour la requete en cours. 
Deux liens qui nous ont aidé à comprendre : 
https://github.com/axios/axios/issues/209
https://stackoverflow.com/questions/41163800/how-does-one-set-global-headers-in-axios